mitodl/odl-video-service
# Upload to youtube via daily celery task instead of signal

#### What are the relevant tickets?
Closes #470 

#### What's this PR do?
Removes signals to upload videos to YouTube.  Adds a daily celery task to sequentially upload public videos until either the maximum limit (`YT_DAILY_UPLOAD_LIMIT`) is reached or the API quota is exceeded, whatever comes first.  An exception will be logged if the quota is reached, to indicate that `YT_DAILY_UPLOAD_LIMIT` should be lowered.

#### How should this be manually tested?
- Set some of your videos to `is_public` in the admin console (they should also have status of Complete)
- In a shell,  run:
  ```python
   from cloudsync.tasks import upload_youtube_videos
   upload_youtube_videos.delay()
    ```
- Videos should start uploading to YouTube
- I can lower the daily quota to a very low number if you want to test the quota limit.
